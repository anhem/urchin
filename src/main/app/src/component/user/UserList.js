import React, {Component} from 'react'

class UserList extends Component {

    render() {
        return(
            <div>hello</div>
        )
    }
}

export default UserList