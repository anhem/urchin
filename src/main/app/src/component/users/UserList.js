import React, {Component} from "react";

class UserList extends Component {

    render() {
        let users = this.props.users || [];
        let userItems = users.map((item, index) =>
            <li key={index.toString()}>
                <a href={`/users/${item.userId}`}>{item.username}</a>
            </li>
        );
        return (
            <div id="user-list">
                <a id="user-list__new-user" href="/users/new">new user</a>
                <h2>Users</h2>
                <ul>
                    {userItems}
                </ul>
            </div>
        )
    }
}

export default UserList