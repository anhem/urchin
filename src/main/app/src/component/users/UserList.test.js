import React from 'react';
import {mount} from 'enzyme'
import UserList from "./UserList";

describe('UserList', () => {

    it('renders without crashing', () => {
        expect(mount(<UserList/>).length).toEqual(1);
    });
});
