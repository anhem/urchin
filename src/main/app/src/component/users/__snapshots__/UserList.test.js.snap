// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserList match snapshot 1`] = `
<div
  id="user-list"
>
  <a
    href="/users/new"
    id="user-list__new-user"
  >
    new user
  </a>
  <h2>
    Users
  </h2>
  <ul />
</div>
`;
